admin-perspective/olm/install-forcenamespace/install-forcenamespace.md

```diff
@@ -0,0 +1,43 @@
+# Operator-defined Namespace
+
+Operator bundles can define what namespace they should be installed to, so that namespace is recommended to the user. The user can still change the installation namespace if desired.
+
+The installed namespace is always conveyed to the user for understanding of the underlying behavior.
+
+## No operator-defined namespace (existing behavior)
+
+![Operator namespace not defined all](img/1-1-notarget-all.png)
+- Namespace selection is read-only and fixed to openshift-operators.
+
+![Operator namespace not defined single](img/1-2-notarget-single.png)
+- All operators are available for installation.
+
+## Operator-defined namespace - already exists
+
+![Operator namespace defined all](img/2-1-targetExists-all.png)
+- Namespace selection defaults to operator-defined namespace with option to change to openshift-operators.
+
+![Operator namespace defined single](img/2-2-targetExists-single.png)
+- Namespace selection defaults to operator-defined namespace with option to change to any other namespace.
+
+## Operator-defined namespace - does not exist
+
+![Operator namespace not created all](img/3-1-targetNotExist-all.png)
+- Namespace selection defaults to operator-defined namespace with option to change to openshift-operators.
+- When the not yet created operator-defined namespace is selected, the user is alerted that the namespace will be created.
+
+![Operator namespace not created single](img/3-2-targetNotExist-single.png)
+- Namespace selection defaults to operator-defined namespace with option to change to any other namespace
+- When the not yet created operator-defined namespace is selected, the user is alerted that the namespace will be created.
+
+## Operator-defined namespace - does not exist, monitoring recommended
+
+![Operator namespace not created monitor all](img/4-1-targetNotExistMonitor-all.png)
+- Namespace selection defaults to operator-defined namespace with option to change to openshift-operators.
+- When the not yet created operator-defined namespace is selected, the user is alerted that the namespace will be created.
+- Checkbox toggle is presented to enable monitoring on newly created namespace (selected by default.)
```

Comment:
should not be checked by default and needs significant warning text around the implications of enabling cluster monitoring on a namespace.

/cc @s-urbaniak @LiliC @brancz

Reply:
Will leave the "(recommended)" text but not check by default. @bparees What are some of the implications that should be noted in the text? I will take a stab at some updated warning text.

Reply:
the implication is that any operator(or other workload) running in an "openshift-*" namespace that is labeled for cluster-monitoring, can contribute its own metrics to the cluster metric set.  That also means they can overwrite existing metrics (they can claim to be reporting pod memory usage, for example, but supply their own bogus values).  So if you're going to let operators/workloads install to those namespaces you need to be sure you trust the metrics they are going to be reporting and there is no risk the component is going to do something nefarious to corrupt your metric values.

Reply:
Would this checkbox still appear if the targetNamespace is not an openshift-* ? Also, is it possible that the targetNamespace is openshift-* but is not recommending for monitoring to be enabled? Would this option still appear just not be recommended?

Assuming it is openshift-* and monitoring recommended, maybe:

[ ] Enable monitoring on this namespace (recommended)
     <sub>Note: Enabling monitoring will allow any operator or workload running on this namespace to contribute metrics to the cluster metric set.</sub>

Reply:
good question. @s-urbaniak @LiliC is any labeling required to enable user workload monitoring on a workspace?  I think the answer was no.

In which case this checkbox really only makes sense as a way to turn on "cluster monitoring" and only makes sense when the namespace in question is an openshift-* one right now.  So maybe it should be greyed out if the namespace is not "openshift-*" or one of the other few specific namespaces that cluster-monitoring is capable of being enabled for.

Reply:
>  is any labeling required to enable user workload monitoring on a workspace?

Right now while its still tech preview, you need to specifically enable user workload monitoring, but otherwise it works.

Reply:
One more thing to note is that, we want to write that this voids support, at least from monitoring side, if the user enables monitoring in a openshift-* namespace and then installs other non red hat operators in that namespace. We should make that a bit warning IMO. Because from what I can tell we are not forcefully preventing this from happening, correct?

Reply:
> Right now while its still tech preview, you need to specifically enable user workload monitoring, but otherwise it works.

ok, but assuming once we go GA the label will not be needed, it seems like this checkbox is really only about enabling cluster monitoring so it should make that clear.

Reply:
> if the user enables monitoring in a openshift-* namespace and then installs other non red hat operators in that namespace

But can we tell which operator is non-redhat ?

Reply:
> But can we tell which operator is non-redhat ? 

no

> Or should we show the explcit warning in all the cases ?

yes

Reply:
> > Or should we show the explcit warning in all the cases ?
> 
> yes

Are we suggesting to warn when monitoring is enabled on openshift-* namespaces? Or are we suggesting to warn when installing any operator to openshift-* namespaces (that monitoring is enabled on)?

What would the warning say?

> > Right now while its still tech preview, you need to specifically enable user workload monitoring, but otherwise it works.
> 
> ok, but assuming once we go GA the label will not be needed, it seems like this checkbox is really only about enabling cluster monitoring so it should make that clear.

I am going forward with the assumption this PR only involves enabling cluster monitoring (installed on openshift-* (including operators), kube-*, default, logging namespace.)

Reply:
> Are we suggesting to warn when monitoring is enabled on openshift-* namespaces? Or are we suggesting to warn when installing any operator to openshift-* namespaces (that monitoring is enabled on)?

The former.

> What would the warning say?

i think we already have warning text in @jhadvig's implementation PR?  we can iterate on it there.